library/std/src/panic.rs

```diff
@@ -17,6 +17,36 @@ use crate::sync::{Arc, Mutex, RwLock};
 use crate::task::{Context, Poll};
 use crate::thread::Result;
 
+#[doc(hidden)]
+#[unstable(feature = "edition_panic", issue = "none", reason = "use panic!() instead")]
+#[allow_internal_unstable(libstd_sys_internals)]
+#[cfg_attr(not(test), rustc_diagnostic_item = "std_panic_2015_macro")]
+#[rustc_macro_transparency = "semitransparent"]
+pub macro panic_2015 {
+    () => ({
+        $crate::rt::begin_panic("explicit panic")
+    }),
+    ($msg:expr $(,)?) => ({
+        $crate::rt::begin_panic($msg)
+    }),
+    ($fmt:expr, $($arg:tt)+) => ({
+        $crate::rt::begin_panic_fmt(&$crate::format_args!($fmt, $($arg)+))
+    }),
+}
+
+#[doc(hidden)]
+#[unstable(feature = "edition_panic", issue = "none", reason = "use panic!() instead")]
+#[allow_internal_unstable(libstd_sys_internals)]
+#[rustc_macro_transparency = "semitransparent"]
+pub macro panic_2021 {
+    () => ({
```

Comment:
The braces are not necessary on 2021 edition (https://github.com/rust-lang/rust/pull/80848#issuecomment-766196634).

Reply:
Also, `std::panic_2021` can simply reexport `core::panic_2021`, and it can do it right now because the issue with braces (https://github.com/rust-lang/rust/pull/80848, https://github.com/rust-lang/rust/issues/80846) is irrelevant to the 2021 version of panic.

Reply:
Yes, that's what I'm doing in https://github.com/rust-lang/rust/commit/c5273bdfb266c35e8eab9413aa8d58d27fdbe114, which I left for a separate PR until we decide on what to do with those braces.

But I agree we can probably just break that in the new edition, even though we cannot easily provide a migration lint for it, as the cases in which the difference is noticable are likely very rare. I'll add that commit in this PR and close the other one.